klippy/kinematics/polar.py

```diff
@@ -130,6 +130,12 @@ def move(self, print_time, move):
                 stepper_bed.set_commanded_position(angle - 2. * math.pi)
         if axes_d[2]:
             self.rails[1].step_itersolve(cmove)
+    def get_status(self):
+        return {'homed_axes': "".join([a for a, h in zip(
+                                   "XYZ", [self.limit_xy2 != -1.] * 2 +
+                                          [self.limit_z[0] <= self.limit_z[1]])
+                                    if h])
```

Comment:
I'm struggling to read that - how about: `return {'homed_axes': ("XY" if self.limit_xy2 >= 0. else "") + ("Z" if self.limit_z[0] <= self.limit_z[1] else "")}`

Reply:
Youre right, when we're returning strings your solution is much easier!